src/containerbuddy/config.go

```diff
@@ -63,6 +66,10 @@ func (s *ServiceConfig) MarkForMaintenance() {
 	s.discoveryService.MarkForMaintenance(s)
 }
 
+func (s *ServiceConfig) Deregister() {
```

Comment:
I like that you've extended the `DiscoveryService` such that if `MarkForMaintenance` and `Deregister` need to be separate implementations for a non-Consul discovery backend we can do that easily.